bundles/org.openhab.binding.revogismartstripcontrol/src/main/java/org/openhab/binding/revogismartstripcontrol/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,168 @@
+
+    private final Logger logger = LoggerFactory.getLogger(RevogiSmartStripControlHandler.class);
+    private final StatusService statusService;
+    private final SwitchService switchService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable RevogiSmartStripControlConfiguration config;
+
+    public RevogiSmartStripControlHandler(Thing thing, StatusService statusService, SwitchService switchService) {
+        super(thing);
+        this.statusService = statusService;
+        this.switchService = switchService;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case PLUG_1_SWITCH:
+                switchPlug(command, 1);
+                break;
+            case PLUG_2_SWITCH:
+                switchPlug(command, 2);
+                break;
+            case PLUG_3_SWITCH:
+                switchPlug(command, 3);
+                break;
+            case PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        if (config == null) {
+            logger.error("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(config.getSerialNumber(), config.ipAddress, port, state);
+        }
+    }
+
+    private int convertOnOffTypeToState(Command command) {
+        if (command == OnOffType.ON) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(RevogiSmartStripControlConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(this::updateStripInformation);
+        Runnable runnable = RevogiSmartStripControlHandler.this::updateStripInformation;
+
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, config.getPollInterval(), TimeUnit.SECONDS);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void updateStripInformation() {
+        if (config == null) {
+            logger.error("No config available, config object was null");
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. In error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.